homeassistant/components/openweathermap/config_flow.py

```diff
@@ -54,14 +55,10 @@ async def async_step_user(self, user_input=None):
             self._abort_if_unique_id_configured()
 
             try:
-                api_online = await _is_owm_api_online(
-                    self.hass, user_input[CONF_API_KEY], latitude, longitude
-                )
-                if not api_online:
+                api_key_valid = await _is_owm_api_key_valid(user_input[CONF_API_KEY])
+                if api_key_valid is False:
                     errors["base"] = "invalid_api_key"
```

Comment:
```
                if api_key_valid is False:
                    errors["base"] = "invalid_api_key"
```

This block cannot raise any of the exceptions being caught, please move it out of the try

Reply:
done